### PR TITLE
api(trace): change SpanID to byte array

### LINKED ISSUE #239: Change SpanID to a byte array
This is a follow up from #183.

We changed the `TraceID` to a byte array because it is more intuitive and easier to use. We should do it to `SpanID` as well.

### PR DESCRIPTION
**Pull request by paivagustavo**
Fixes #239

### CODE REVIEW

File: api/core/span_context.go [hunk 1]
```diff
@@ -49,72 +68,82 @@ func TraceIDFromHex(h string) (TraceID, error) {
...
+		return t, err
+	}
+
+	if !t.IsValid() {
+		return t, errors.New("trace-id can't be all zero")
+	}
+	return t, nil
+}
+
+// TraceIDFromHex returns a SpanID from a hex string if it is compliant
```
Comment (rghetia) [inline api/core/span_context.go]: s/TraceIDFromHex/SpanIDFromHex/

File: api/core/span_context.go [hunk 2]
```diff
@@ -49,72 +68,82 @@ func TraceIDFromHex(h string) (TraceID, error) {
...
+// with the w3c trace-context specification.
+// See more at https://www.w3.org/TR/trace-context/#parent-id
+func SpanIDFromHex(h string) (SpanID, error) {
+	s := SpanID{}
+	if len(h) != 16 {
+		return s, errors.New("hex encoded span-id must have length equals to 16")
```
Comment (rghetia) [inline api/core/span_context.go]: make this and other errors  a static error.

File: api/core/span_context.go (hunk 1 above)
Comment (paivagustavo) [reply api/core/span_context.go]: ops, changed.

File: api/core/span_context.go (hunk 2 above)
Comment (paivagustavo) [reply api/core/span_context.go]: changed

**Pull request merged**